lib/java/src/org/apache/thrift/async/TAsyncMethodCall.java

```diff
@@ -225,8 +231,13 @@ private void cleanUpAndFireCallback(SelectionKey key) {
     key.interestOps(0);
     // this ensures that the TAsyncMethod instance doesn't hang around
     key.attach(null);
```

Comment:
maybe it's just a bug in the original code ?
Then I think we can simply add cancel() and stop key set from growing indefinitely without doing any harm at all.
What do you think ?